### PR TITLE
Initial

### PR DESCRIPTION
**Pull request by remi-dupre**

### CODE REVIEW

File: pyproject.toml [hunk 1]
```diff
@@ -0,0 +1,4 @@
+[tool.black]
+line-length = 79
+target-version = ['py37']
+skip-string-normalization = true
```
Comment (jbgriesner) [inline pyproject.toml:4]: Is that file necessary ?

File: src/config.py [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+    '{year:04d}/{year:04d}-{month:02d}/'
+    'pageviews-{year:04d}{month:02d}{day:02d}-{hour:02d}0000.gz'.format(
+        year=2019, month=9, day=day, hour=hour
+    )
+    for day in range(1, 30)
+    for hour in range(24)
+]
+
+# Sites we want statistics for
+STATS_SITES = ['fr', 'en', 'de', 'ja']
```
Comment (jbgriesner) [inline src/config.py]: What about putting these configs in a .yml file ?

File: src/download_stats.py [hunk 3]
```diff
@@ -0,0 +1,16 @@
+#!/usr/bin/env python3
+import os
+import urllib.request
+
+from config import *
+
+
+def download_stats():
```
Comment (jbgriesner) [inline src/download_stats.py:9]: Many exceptions could be raised here but won't be caught

File: src/main.py [hunk 4]
```diff
@@ -0,0 +1,77 @@
...
+import argparse
+import sys
+
+from download_stats import download_stats
+from load_stats import load_stats
+from load_wikidata import load_wikidata
+
+
+#  Read CLI arguments
+parser = argparse.ArgumentParser(
```
Comment (jbgriesner) [inline src/main.py:12]: You could probably use more abstractions here to handle the arguments

**Review by jbgriesner — APPROVED** [review]

File: src/download_stats.py [hunk 5]
```diff
@@ -0,0 +1,16 @@
...
+    for i, dump_file in enumerate(STATS_FILES):
```
Comment (amatissart) [inline src/download_stats.py]: `STATS_FILES` does not exist

File: src/download_stats.py [hunk 6]
```diff
@@ -0,0 +1,16 @@
+#!/usr/bin/env python3
+import os
+import urllib.request
+
+from config import *
```
Comment (amatissart) [inline src/download_stats.py]: `import *` is often considered a bad practice in python, as it may hide invalid imports, wrong variables, etc.

File: src/load_stats.py [hunk 7]
```diff
@@ -0,0 +1,55 @@
...
+        path = os.path.join(STATS_DUMP_DIR, dump_file)
+        print('({}/{}) {}'.format(i + 1, len(STATS_DUMP_FILES), path))
+
+        for i, line in enumerate(gzip.open(path)):
+            try:
+                site, page, views, responses = line.decode().strip().split()
+            except Exception as e:
+                print('Error at line', i, ':', e, file=sys.stderr)
+
+            page = page.replace('_', ' ')
```
Comment (amatissart) [inline src/load_stats.py]: in case of an error, `page` may be undefined here

Comment (remi-dupre) [reply src/load_stats.py]: I fixed my linter's configuration

**Review by amatissart — APPROVED** [review]
